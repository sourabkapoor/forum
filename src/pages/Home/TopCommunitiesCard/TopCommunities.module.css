.top-communities {
  display: grid;
  place-items: center;
  padding: 20px 24px;
  width: 100%;
}
.top-communities__content {
  display: flex;
}
.top-communities__posts {
}

.top-communities__top-community-image {
  width: 100%;
  /* background-image: linear-gradient(
      to bottom,
      rgba(0, 0, 0, 0),
      rgba(0, 0, 0, 0.75)
    ),
    url(../../../assets/communities.png); */
  border-top-left-radius: 4px;
  border-top-right-radius: 4px;
  display: flex;
  justify-content: flex-start;
  align-items: flex-end;
}

.top-communities__top-community-title {
  color: Black;
  font-weight: 600;
  padding-left: 0.5rem;
  padding-bottom: 0.5rem;
}
.top-communities__top-community-list {
  padding: 0;
}

.top-communities__top-community-item {
  display: flex;
  justify-content: flex-start;
  align-items: center;
  border-radius: 6px;
  height: 48px;
  cursor: pointer;
}

.top-communities__top-community-item:hover {
  background-color: #e8bff7;
}

.top-communities__top-community-details {
  display: flex;
  margin: 0 12px;
  gap: 8px;
}

.top-communities__top-comunity-rank {
  color: white;
  text-align: right;
  min-width: 20px;
}
.top-communities__top-community-icon {
  filter: invert(48%) sepia(79%) saturate(2476%) hue-rotate(86deg)
    brightness(118%) contrast(119%);
  width: 20px;
  height: 20px;
}
.top-communities__top-comunity-name {
  font-size: 14px;
  font-weight: 500;
  line-height: 18px;
}

.top-communities__top-community-button {
  background-color: white;
  border: none;
  color: var(--primary-secondary);
  font-weight: 700;
  letter-spacing: unset;
  line-height: 16px;
  text-transform: unset;
  color: black;
  border-radius: 9999px;
  box-sizing: border-box;
  cursor: pointer;
}

.top-communities__skeletons {
  display: flex;
  flex-direction: column;
}
.top-communities__top-community-button_type_join {
  min-height: 24px;
  min-width: 24px;
  padding: 4px 16px;
}
.top-communities__top-community-button_type_view {
  width: 100%;
  height: 32px;
}
.top-communities__top-community-button:hover {
  background-color: var(--secondary-color);
  cursor: pointer;
}
