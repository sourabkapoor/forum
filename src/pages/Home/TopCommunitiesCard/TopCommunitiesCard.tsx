import CSSModules from "react-css-modules";
import Card from "../../../components/Card/Card";
import styles from "./TopCommunities.module.css";
import { IoIosArrowUp } from "react-icons/io";
import { useEffect, useState } from "react";
import { db } from "../../../firebase";
import {
  collection,
  DocumentData,
  getDocs,
  limit,
  orderBy,
  query,
  where,
} from "firebase/firestore";
import { Link, useNavigate } from "react-router-dom";
import UpworkJobLoader from "../../../components/Skeletons/UpworkJobLoader";

const TopCommunitiesCard: React.FC = () => {
  const [topCommunities, setTopCommunities] = useState<
    DocumentData | undefined
  >();

  const navigate = useNavigate();

  useEffect(() => {
    async function fetchTopCommunities() {
      try {
        const subredditsRef = collection(db, "subreddits");

        const q = query(
          subredditsRef,
        );

        const subredditsDocs = await getDocs(q);

        console.log(subredditsDocs.docs);
        const filteredDocs = subredditsDocs.docs.filter((doc: DocumentData) =>
          doc.data().hasOwnProperty("name")
        );
        setTopCommunities(filteredDocs);
      } catch (error) {
        console.log(`Could not fetch top communities: ${error}`);
      }
    }

    fetchTopCommunities();
  }, []);

  return (
    <div style={{padding: '24px', width: "100%"}}>
      <Card>
        <div styleName="top-communities__top-community-image">
          <h2 styleName="top-communities__top-community-title">
            All Communities
          </h2>
        </div>
        <ol styleName="top-communities__top-community-list">
          {topCommunities?.map((doc: DocumentData, index: number) => {
            return (
              <li key={doc.id} styleName="top-communities__top-community-item">
                <Link to={`/r/${doc.data().name}`} style={{width: "100%"}}>
                  <div styleName="top-communities__top-community-details">
                    <p styleName="top-communities__top-community-rank">
                      {index + 1}
                    </p>
                    <span styleName="top-communities__top-community-name">
                      {doc.data().name}
                    </span>
                  </div>
                </Link>
              </li>
            );
          }) ?? (
            <div styleName="top-communities__skeletons">
              <UpworkJobLoader
                animate={true}
                width={321}
                backgroundColor={"#333"}
                foregroundColor={"#999"}
                speed={1}
              />
              <UpworkJobLoader
                animate={true}
                width={321}
                backgroundColor={"#333"}
                foregroundColor={"#999"}
                speed={1}
              />
              <UpworkJobLoader
                animate={true}
                width={321}
                backgroundColor={"#333"}
                foregroundColor={"#999"}
                speed={1}
              />
            </div>
          )}
        </ol>
      </Card>
    </div>
  );
};

export default CSSModules(TopCommunitiesCard, styles, {
  allowMultiple: true,
  handleNotFoundStyleName: "ignore",
});
