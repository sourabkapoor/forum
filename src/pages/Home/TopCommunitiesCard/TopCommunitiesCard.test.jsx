import { queryByPlaceholderText, render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import userEvent from "@testing-library/user-event";
import each from "jest-each";
import TopCommunitiesCard from "./TopCommunitiesCard";

describe("TopCommunitiesCard", () => {
  test("snapshot", () => {
    const { asFragment } = render(<TopCommunitiesCard />);

    expect(asFragment()).toMatchSnapshot();
  });
});
