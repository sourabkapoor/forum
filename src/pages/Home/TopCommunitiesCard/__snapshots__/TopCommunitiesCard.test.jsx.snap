// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopCommunitiesCard snapshot 1`] = `
<DocumentFragment>
  <div
    class="card"
  >
    <div
      class="top-communities__top-community-image"
    >
      <h2
        class="top-communities__top-community-title"
      >
        Top Communities
      </h2>
    </div>
    <ol
      class="top-communities__top-community-list"
    >
      <li
        class="top-communities__top-community-item"
      >
        <a
          class="top-communities__top-community-details"
        >
          <p
            class="top-communities__top-community-rank"
          >
            1
          </p>
          <svg
            class="top-communities__top-community-icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 217.9L383 345c9.4 9.4 24.6 9.4 33.9 0 9.4-9.4 9.3-24.6 0-34L273 167c-9.1-9.1-23.7-9.3-33.1-.7L95 310.9c-4.7 4.7-7 10.9-7 17s2.3 12.3 7 17c9.4 9.4 24.6 9.4 33.9 0l127.1-127z"
            />
          </svg>
          <span
            class="top-communities__top-community-name"
          >
            r/gaming
          </span>
        </a>
        <button
          class="top-communities__top-community-button top-communities__top-community-button_type_join"
        >
          Join
        </button>
      </li>
      <li
        class="top-communities__top-community-item"
      >
        <a
          class="top-communities__top-community-details"
        >
          <p
            class="top-communities__top-community-rank"
          >
            1
          </p>
          <svg
            class="top-communities__top-community-icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 217.9L383 345c9.4 9.4 24.6 9.4 33.9 0 9.4-9.4 9.3-24.6 0-34L273 167c-9.1-9.1-23.7-9.3-33.1-.7L95 310.9c-4.7 4.7-7 10.9-7 17s2.3 12.3 7 17c9.4 9.4 24.6 9.4 33.9 0l127.1-127z"
            />
          </svg>
          <span
            class="top-communities__top-community-name"
          >
            r/gaming
          </span>
        </a>
        <button
          class="top-communities__top-community-button top-communities__top-community-button_type_join"
        >
          Join
        </button>
      </li>
      <li
        class="top-communities__top-community-item"
      >
        <a
          class="top-communities__top-community-details"
        >
          <p
            class="top-communities__top-community-rank"
          >
            1
          </p>
          <svg
            class="top-communities__top-community-icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 217.9L383 345c9.4 9.4 24.6 9.4 33.9 0 9.4-9.4 9.3-24.6 0-34L273 167c-9.1-9.1-23.7-9.3-33.1-.7L95 310.9c-4.7 4.7-7 10.9-7 17s2.3 12.3 7 17c9.4 9.4 24.6 9.4 33.9 0l127.1-127z"
            />
          </svg>
          <span
            class="top-communities__top-community-name"
          >
            r/gaming
          </span>
        </a>
        <button
          class="top-communities__top-community-button top-communities__top-community-button_type_join"
        >
          Join
        </button>
      </li>
      <li
        class="top-communities__top-community-item"
      >
        <a
          class="top-communities__top-community-details"
        >
          <p
            class="top-communities__top-community-rank"
          >
            1
          </p>
          <svg
            class="top-communities__top-community-icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 217.9L383 345c9.4 9.4 24.6 9.4 33.9 0 9.4-9.4 9.3-24.6 0-34L273 167c-9.1-9.1-23.7-9.3-33.1-.7L95 310.9c-4.7 4.7-7 10.9-7 17s2.3 12.3 7 17c9.4 9.4 24.6 9.4 33.9 0l127.1-127z"
            />
          </svg>
          <span
            class="top-communities__top-community-name"
          >
            r/gaming
          </span>
        </a>
        <button
          class="top-communities__top-community-button top-communities__top-community-button_type_join"
        >
          Join
        </button>
      </li>
      <li
        class="top-communities__top-community-item"
      >
        <a
          class="top-communities__top-community-details"
        >
          <p
            class="top-communities__top-community-rank"
          >
            1
          </p>
          <svg
            class="top-communities__top-community-icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 217.9L383 345c9.4 9.4 24.6 9.4 33.9 0 9.4-9.4 9.3-24.6 0-34L273 167c-9.1-9.1-23.7-9.3-33.1-.7L95 310.9c-4.7 4.7-7 10.9-7 17s2.3 12.3 7 17c9.4 9.4 24.6 9.4 33.9 0l127.1-127z"
            />
          </svg>
          <span
            class="top-communities__top-community-name"
          >
            r/gaming
          </span>
        </a>
        <button
          class="top-communities__top-community-button top-communities__top-community-button_type_join"
        >
          Join
        </button>
      </li>
    </ol>
    <button
      class="top-communities__top-community-button top-communities__top-community-button_type_view"
    >
      View All
    </button>
  </div>
</DocumentFragment>
`;
